src/gui/src/MainForm.cpp

```diff
@@ -1347,50 +1481,27 @@ void MainForm::updateRecentUsedSongList()
 
 void MainForm::action_file_open_recent(QAction *pAction)
 {
-	//	INFOLOG( pAction->text() );
-	openSongFile( pAction->text() );
-}
-
-
-
-void MainForm::openSongFile( const QString& sFilename )
-{
-	Hydrogen *pEngine = Hydrogen::get_instance();
-	if ( pEngine->getState() == STATE_PLAYING ) {
-		pEngine->sequencer_stop();
+#ifdef H2CORE_HAVE_OSC
```

Comment:
I'm wondering if we could get rid of many of those preprocessor magic if we would put the preprocesser macros inside the NSM class iso. of around its uses. Would this be possible? I did sth. similar with the Ladspa includes in my lv2 playground branch. This had the advantage that the compiler and static analysis tool always see as much as code as possible and would problems with variable renaming in those sections.

 Of course that might mean that NsmClient::get_instance() might be a nullptr or that the NsmClient class itself would have an "enabled" flag or sth. like that.

Reply:
sure